src/test/java/de/tum/in/www1/artemis/hestia/HestiaDatabaseTest.java

```diff
@@ -0,0 +1,197 @@
+
+    @Autowired
+    private ProgrammingExerciseTestCaseRepository programmingExerciseTestCaseRepository;
+
+    @Autowired
+    private ProgrammingExerciseTaskRepository programmingExerciseTaskRepository;
+
+    @Autowired
+    private ProgrammingExerciseSolutionEntryRepository programmingExerciseSolutionEntryRepository;
+
+    @Autowired
+    private CodeHintRepository codeHintRepository;
+
+    private Long programmingExerciseId;
+
+    @BeforeEach
+    void init() {
+        database.addUsers(2, 2, 0, 2);
+        database.addCourseWithOneProgrammingExercise();
+        programmingExerciseId = programmingExerciseRepository.findAll().get(0).getId();
+    }
+
+    @AfterEach
+    void tearDown() {
+        database.resetDatabase();
+    }
+
+    ProgrammingExerciseTask addTaskToProgrammingExercise(String taskName) {
+        var task = new ProgrammingExerciseTask();
+        task.setTaskName(taskName);
+        task.setExercise(programmingExerciseRepository.getById(programmingExerciseId));
+        task = programmingExerciseTaskRepository.save(task);
+        return task;
+    }
+
+    ProgrammingExerciseSolutionEntry[] addSolutionEntriesToTestCase(int count, ProgrammingExerciseTestCase testCase) {
+        var solutionEntries = new ProgrammingExerciseSolutionEntry[count];
+        for (int i = 0; i < count; i++) {
+            var solutionEntry = new ProgrammingExerciseSolutionEntry();
+            solutionEntry.setTestCase(testCase);
+            solutionEntry.setCode("Code block 1");
+            solutionEntry.setLine(i);
+            solutionEntry = programmingExerciseSolutionEntryRepository.save(solutionEntry);
+            solutionEntries[i] = solutionEntry;
+        }
+        return solutionEntries;
+    }
+
+    @Test
+    void addOneTaskToProgrammingExercise() {
+        var task = addTaskToProgrammingExercise("Task 1");
+        assertThat(programmingExerciseTaskRepository.findByExerciseIdWithTestCases(programmingExerciseId)).containsExactly(task);
+    }
+
+    @Test
+    void deleteProgrammingExerciseWithTask() {
+        addOneTaskToProgrammingExercise();
+        programmingExerciseRepository.deleteById(programmingExerciseId);
+        assertThat(programmingExerciseTaskRepository.findAll()).isEmpty();
+    }
+
+    @Test
+    void addTestCasesWithSolutionEntriesToProgrammingExercise() {
+        var programmingExercise = programmingExerciseRepository.getById(programmingExerciseId);
+        database.addTestCasesToProgrammingExercise(programmingExercise);
+        var testCases = programmingExerciseTestCaseRepository.findByExerciseId(programmingExerciseId);
+        assertThat(testCases).isNotEmpty();
+        for (ProgrammingExerciseTestCase testCase : testCases) {
+            var solutionEntries = addSolutionEntriesToTestCase(2, testCase);
+            assertThat(programmingExerciseSolutionEntryRepository.findByTestCaseId(testCase.getId())).hasSize(2).containsExactly(solutionEntries);
```

Comment:
```suggestion
            assertThat(programmingExerciseSolutionEntryRepository.findByTestCaseId(testCase.getId())).containsExactly(solutionEntries);
```
I think the `hasSize` assertion is unnecessary if you then check for the exact contents anyway. (Multiple occurrences)